src/assets/rewardrequestdb.cpp

```diff
@@ -0,0 +1,282 @@
+// Copyright (c) 2019 The Raven Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <set>
+
+#include "rewardrequestdb.h"
+
+static const char SCHEDULEDREWARD_FLAG = 'S';
+
+CRewardRequest::CRewardRequest()
+{
+    SetNull();
+}
+
+CRewardRequest::CRewardRequest(
+    const std::string & p_rewardID,
+    const std::string & p_walletName, const int p_heightForPayout, const CAmount p_totalPayoutAmt,
+    const std::string & p_payoutSrc, const std::string & p_tgtAssetName, const std::string & p_exceptionAddresses
+)
+{
+    SetNull();
+
+    rewardID = p_rewardID;
+    walletName = p_walletName;
+    heightForPayout = p_heightForPayout;
+    totalPayoutAmt = p_totalPayoutAmt;
+    payoutSrc = p_payoutSrc;
+    tgtAssetName = p_tgtAssetName;
+    exceptionAddresses = p_exceptionAddresses;
+}
+
+CRewardRequestDBEntry::CRewardRequestDBEntry()
+{
+    SetNull();
+}
+
+CRewardRequestDBEntry::CRewardRequestDBEntry(
+    const std::set<CRewardRequest> & p_requests
+)
+{
+    SetNull();
+
+    for (auto const & currReq : p_requests) {
+        requests.insert(currReq);
+    }
+}
+
+CRewardRequestDB::CRewardRequestDB(size_t nCacheSize, bool fMemory, bool fWipe) : CDBWrapper(GetDataDir() / "rewardrequest", nCacheSize, fMemory, fWipe) {
+}
+
+bool CRewardRequestDB::SchedulePendingReward(const CRewardRequest & p_newReward)
+{
+    LogPrintf("%s : Scheduling reward: rewardID='%s', wallet='%s', heigh t=%d, amt=%lld, srcAsset='%s', tgtAsset='%s', exceptions='%s'\n",
+        __func__,
+        p_newReward.rewardID.c_str(),
+        p_newReward.walletName.c_str(), p_newReward.heightForPayout, static_cast<long long>(p_newReward.totalPayoutAmt),
+        p_newReward.payoutSrc.c_str(), p_newReward.tgtAssetName.c_str(),
+        p_newReward.exceptionAddresses.c_str());
+
+    //  Retrieve height-based entry which contains list of rewards to pay at that height
+    CRewardRequestDBEntry reqDbEntry;
+
+    //  Result doesn't matter, as this might be the first entry added at this height
+    if (!Read(std::make_pair(SCHEDULEDREWARD_FLAG, p_newReward.heightForPayout), reqDbEntry)) {
+        LogPrintf("%s : Failed to read entry for height %d!\n",
+            __func__,
+            p_newReward.heightForPayout);
+    }
+
+    reqDbEntry.requests.insert(p_newReward);
+
+    for (auto const & reward : reqDbEntry.requests) {
+        LogPrintf("%s : Reward at height %d: rewardID='%s', wallet='%s', amt=%lld, srcAsset='%s', tgtAsset='%s', exceptions='%s'\n",
+            __func__,
+            reward.heightForPayout, reward.rewardID.c_str(),
+            reward.walletName.c_str(), static_cast<long long>(reward.totalPayoutAmt),
+            reward.payoutSrc.c_str(), reward.tgtAssetName.c_str(),
+            reward.exceptionAddresses.c_str());
+    }
```

Comment:
Need to make sure this doesn't make it passed testing. 
Could be good to put a ```//TODO - Remove this log loop before releasing```

Reply:
This block has been removed.